pkg/security/podsecuritypolicy/provider.go

```diff
@@ -449,3 +452,13 @@ func validateRuntimeClassName(actual *string, validNames []string) field.ErrorLi
 	}
 	return field.ErrorList{field.Invalid(field.NewPath("spec", "runtimeClassName"), *actual, "")}
 }
+
+func allowsVolumeType(allowedVolumes sets.String, fsType policy.FSType, volume api.Volume) bool {
+	if allowedVolumes.Has(string(fsType)) {
+		return true
+	}
+	if allowedVolumes.Has(string(policy.Secret)) && fsType == policy.Projected && utilfeature.DefaultFeatureGate.Enabled(features.BoundServiceAccountTokenVolume) && reflect.DeepEqual(*volume.VolumeSource.Projected, *serviceaccount.TokenVolumeSource) {
```

Comment:
Drop the feature gate check here. Scenario:
1. A new pod gets created against an API server with the feature enabled, and gets a projected serviceaccount token volume injected
2. An update of the existing pod gets handled by an API server with the feature disabled. That API server should still allow the pod based on this logic.

Reply:
ok, the second scenario seems cover the case where the service account token sources are injected manually. i could not find other reasons why api server would allow pods if the feature is off.

Reply:
That's a single scenario consisting of two steps. It must be valid to update an existing pod that had a projected serviceaccount token volume added when it was created, even if the update is done on an API server that does not have the feature enabled.

Reply:
i see.